cmd/skaffold/app/cmd/runner.go

```diff
@@ -149,3 +150,13 @@ func warnIfUpdateIsAvailable() {
 		log.Entry(context.TODO()).Warn(warning)
 	}
 }
+
+func setDefaultRendererAndDeployer(configs parser.SkaffoldConfigSet) {
+	// do not set a default deployer or renderer in a multi-config application.
+	if len(configs) > 1 {
```

Comment:
Verified this behavior here https://github.com/GoogleContainerTools/skaffold/blob/v1/cmd/skaffold/app/cmd/runner.go#L145